float64[2]     parameter         # parameter p
float64[]      initial_guess     # u0
float64[]      initial_y         # y0
float64        initial_penalty
